src/Avalonia.Controls/HyperlinkButton.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using Avalonia.Controls.Metadata;
+using Avalonia.Threading;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// A button control that functions as a navigateable hyperlink.
+    /// </summary>
+    [PseudoClasses(pcVisited)]
+    public class HyperlinkButton : Button
+    {
+        // See: https://www.w3schools.com/cssref/sel_visited.php
+        private const string pcVisited = ":visited";
+
+        /// <summary>
+        /// Defines the <see cref="IsVisited"/> property.
+        /// </summary>
+        public static readonly StyledProperty<bool> IsVisitedProperty =
+            AvaloniaProperty.Register<HyperlinkButton, bool>(
+                nameof(IsVisited),
+                defaultValue: false);
+
+        /// <summary>
+        /// Defines the <see cref="IsVisitedOnClick"/> property.
+        /// </summary>
+        public static readonly StyledProperty<bool> IsVisitedOnClickProperty =
+            AvaloniaProperty.Register<HyperlinkButton, bool>(
+                nameof(IsVisitedOnClick),
+                defaultValue: true);
+
+        /// <summary>
+        /// Defines the <see cref="NavigateUri"/> property.
+        /// </summary>
+        public static readonly StyledProperty<Uri?> NavigateUriProperty =
+            AvaloniaProperty.Register<HyperlinkButton, Uri?>(
+                nameof(NavigateUri),
+                defaultValue: null);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HyperlinkButton"/> class.
+        /// </summary>
+        public HyperlinkButton()
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the <see cref="NavigateUri"/> has been visited.
+        /// </summary>
+        public bool IsVisited
+        {
+            get => GetValue(IsVisitedProperty);
+            set => SetValue(IsVisitedProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the <see cref="IsVisited"/> property is automatically
+        /// set to true when the <see cref="HyperlinkButton"/> is clicked and the <see cref="NavigateUri"/>
+        /// is successfully launched.
+        /// </summary>
+        public bool IsVisitedOnClick
```

Comment:
The previous attempt used the name `IsVisitedSetOnClick`. I think the `Set` is implied so removed it.

Additionally, there is some debate if the name should be `IsUriVisitedOnClick`. Again, the Uri seems implied but may be better to include it.

Reply:
I am personally in favour of using `IsVisitedSetOnClick`. It sounds far less awkward and is clearer what it is doing. I would also be fine with `SetVisitedOnClick`. That said, I would prefer it not have "click" at all in the name. It conflicts with Avalonia's existing semantics which is pointers.

Reply:
Property removed and I added comments in NavigateUri for proper usage.